src/main/java/seedu/recipe/model/recipe/ingredient/IngredientQuantifier.java

```diff
@@ -0,0 +1,52 @@
+package seedu.recipe.model.recipe.ingredient;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Represents a container for the values stored with respect to an Ingredient,
+ * within a Recipe instance.
+ */
+public class IngredientQuantifier {
+    private final Optional<IngredientQuantity> quantityContainer;
+    private final Optional<String> commonNameContainer;
+    private final List<String> estimatedQuantity = new ArrayList<>();
+    private final List<String> remarks = new ArrayList<>();
+    private final List<Ingredient> substitutions = new ArrayList<>();
```

Comment:
hm I feel like the `commonName` and `substitutions` should belong to `Ingredient` instead of `IngredientQuantifier`? cause those belong to the ingredient itself and is not necessarily recipe-specific

remarks and quantity is fine since they exist on a per-recipe basis (e.g. some recipes ask for roughly-chopped carrots, others ask for it diced or cubed)

Reply:
fixed